### PR TITLE
io: fix stream reentrancy, path handling and temp-file suffix bugs

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary

Cluster E of the project-analysis fix series (findings catalog: `doc/analysis-findings.md`, IDs IO-1..IO-16).

- **IO-1 (high)**: `UrlLoader` cached its `URLConnection`, and JDK connections hand out the *same* stream instance — so `isReadable()` (which opens and closes the stream) left every subsequent `inputStream()` call with a closed stream ("Stream closed"). This broke the `FileLoader` reentrancy contract and `FileLoaderUtility.copyFileToTemp` for URL sources. Now opens a fresh connection per call.
- **IO-2**: the context-classloader fallback in `ClassPathLoader` passed a leading-slash name to `ClassLoader.getResource()`, which requires slash-less names — the fallback could never succeed. Fixed and pinned with a custom-TCCL test.
- **IO-3**: `MorePaths.backupFile` NPE'd for single-segment relative paths (null parent); now resolves via `toAbsolutePath()`.
- **IO-4**: `deleteQuietly` NPE'd when `File.list()` returned null (I/O error/permission denied), contradicting its "never throws" javadoc.
- **IO-5**: `FileSystemLoader.outputStream()` javadoc claimed it creates missing files — impossible, since `writable` is a construction-time snapshot requiring an existing file (the existing test asserts the opposite of the old doc). Docs now state the real contract.
- **IO-6**: `external:` prefix handling — a swallowed `IOException` let the unprocessed `"external:..."` string flow on as a nonsense path (now fails fast with cause), and the working-directory prefix was concatenated without a separator (`external:conf/app.txt` → `/cwdconf/app.txt`). The prefix is now documented.
- **IO-7**: temp copies passed the suffix without a dot to `createTempFile` (`report…886txt`), mangling the extension consumers dispatch on. Now `"." + suffix`.
- **IO-14**: `saveAndBackup` leaked one temp file per invocation; now cleaned up in a `finally`. Doc referenced nonexistent `PathUtils`.
- **IO-10**: removed `IOCase.readResolve()` and the `transient` modifier — both no-ops for enums (serialized by name).
- **IO-8, IO-9, IO-11..IO-13**: javadoc corrections — package examples called three nonexistent methods; `UrlLoader` class doc described the opposite of its behavior with an inverted `@throws`; wrong exception type on `IOStreams.toString`; "not readable" copy-paste in `FileWriter`; missing leading slash in a `FilenameUtils.concat` example.
- **IO-15/IO-16 (tests)**: `UrlLoader` content reads after `isReadable()` and repeated reads (pins IO-1); first-ever coverage for `IOCase.checkEndsWith` including the longer-suffix edge case; temp-file extension assertions.

## Test plan

- `./mvnw clean install` — BUILD SUCCESS, all tests green (UrlLoaderTest 12/12, IOCaseTest 32/32, MorePathsTest 20/20)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01NKzn4Upj33uRFvWQep3UzE

---
_Generated by [Claude Code](https://claude.ai/code/session_01NKzn4Upj33uRFvWQep3UzE)_

### CODE REVIEW

File: src/test/java/de/cuioss/tools/io/MorePathsTest.java [hunk 1]
```diff
@@ -263,6 +292,11 @@ void deleteQuietlyFile() throws IOException {
 
     }
 
+    @Test
+    void deleteQuietlyShouldNotThrowForEmptyPath() {
+        assertDoesNotThrow(() -> assertFalse(deleteQuietly(Path.of(""))));
+    }
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply src/test/java/de/cuioss/tools/io/MorePathsTest.java:298]: Fixed in 08b9a8c — `deleteQuietly` now explicitly refuses empty paths (`path.toString().isEmpty()` → `false`) before any filesystem interaction, so the empty-path test is safe even if the implementation later moves to NIO.2 `Files.exists`. Good catch.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/main/java/de/cuioss/tools/io/FileSystemLoader.java [hunk 2]
```diff
@@ -122,11 +124,18 @@ public static String checkPathName(final String pathName) {
         if (pathName.startsWith(FileTypePrefix.FILE.getPrefix())) {
             newPathName = FileTypePrefix.FILE.removePrefix(pathName);
         } else if (pathName.startsWith(FileTypePrefix.EXTERNAL.getPrefix())) {
+            var remainder = FileTypePrefix.EXTERNAL.removePrefix(pathName);
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply src/main/java/de/cuioss/tools/io/FileSystemLoader.java:127]: Fixed in 08b9a8c — `"external:"` with an empty remainder now throws `IllegalArgumentException` as the javadoc promises, with a new test `shouldRejectExternalPrefixWithoutPath`.

---
_Generated by [Claude Code](https://claude.ai/code)_
